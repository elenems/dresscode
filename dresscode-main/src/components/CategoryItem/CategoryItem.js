import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
function CategoryItem(props) {
  return (
    <div>
      <Link to={{ pathname: props.link, search: props.search }}>
        <img src={props.image} alt={props.categoryName} />
        <p>{props.categoryName}</p>
      </Link>
    </div>
  );
}

CategoryItem.propTypes = {
  link: PropTypes.string.isRequired,
  categoryName: PropTypes.string.isRequired
};

export default React.memo(CategoryItem);
