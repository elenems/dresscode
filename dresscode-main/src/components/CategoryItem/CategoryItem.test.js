import { shallow } from "enzyme";
import React from "react";
import CategoryItem from "./CategoryItem";

describe("CategoryItem", () => {
  const props = {
    categoryName: "Shoes"
  };
  it("Matches snapshot", () => {
    const wrapper = shallow(<CategoryItem />).debug();
    expect(wrapper).toMatchSnapshot();
  });

  it("Shows category name in paragraph", () => {
    const wrapper = shallow(<CategoryItem {...props} />);
    expect(wrapper.find("p").text()).toBe(props.categoryName);
  });

  it("img has alt attribute", () => {
    const wrapper = shallow(<CategoryItem {...props} />);
    expect(wrapper.find('img').prop('alt')).toBe(props.categoryName)
  });
});
