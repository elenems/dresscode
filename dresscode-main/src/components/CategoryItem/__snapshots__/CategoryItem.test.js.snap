// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryItem Matches snapshot 1`] = `
"<div>
  <Link to={{...}}>
    <img src={[undefined]} alt={[undefined]} />
    <p />
  </Link>
</div>"
`;
